upgrade/v122to123/upgrade.go

```diff
@@ -42,6 +42,24 @@ func upgradeBackups(namespace string, lhClient *lhclientset.Clientset) (err erro
 		return err
 	}
 
+	engines, err := lhClient.LonghornV1beta1().Engines(namespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	volumeNameToEngines := make(map[string][]longhorn.Engine)
```

Comment:
NIT: is it better to use pointers here? `map[string][]*longhorn.Engine`. Similar below

Reply:
Fixed it. I has been always scare of using pointer :D

Reply:
It's better not to the pointers of objects init by the `for` loop. IIRC, the for loop here can be considered as:
```
var e longhorn.Engine
for idx := 0; idx < len(engines.Items); i++ {
    e = engines.Items[idx]
    p := &e  // This always points to the same variable, which may be problematic
}
```

For more detail, you can try this: https://go.dev/play/p/tjB7R-znss6

Reply:
This kind of syntactic sugar is tricky anyway.

Reply:
Screw me! My big mistake.
 Many thanks for the alarm!

Reply:
@shuo-wu 
PR created at https://github.com/longhorn/longhorn-manager/pull/1340